contracts/ReferenceFiefdom.sol

```diff
@@ -13,43 +15,39 @@ interface IBaseContract {
   function defaultTokenURIContract() external view returns (address tokenURIContract);
 }
 
-interface ITokenURI {
-  function tokenURI(uint256 tokenId) external view returns (string memory uri);
-}
-
-contract ReferenceFiefdom is Initializable, ERC721Upgradeable {
+contract ReferenceFiefdom is Initializable, ERC721 {
   using Strings for uint256;
 
   string public license;
+  IBaseContract public overlord;
+  ITokenURI public tokenURIContract;
+  IERC721Hooks public erc721Hooks;
   address public minter;
   uint256 public fiefdom;
-  IBaseContract public overlord;
-  address public tokenURIContract;
   bool public isActivated;
 
   uint256 private _totalSupply;
   uint256 private _maxSupply;
   string private _name;
   string private _symbol;
-
-  address private royaltyBeneficiary;
-  uint16 private royaltyBasisPoints;
+  bool private _isActivated;
+  address private _royaltyBeneficiary;
+  uint16 private _royaltyBasisPoints;
 
   event ProjectEvent(address indexed poster, string indexed eventType, string content);
   event TokenEvent(address indexed poster, uint256 indexed tokenId, string indexed eventType, string content);
 
   // This is only called when the reference contract is published
-  constructor() {
+  constructor() ERC721('', '') {
```

Comment:
we don't actually need this, do we?

Reply:
We do, it can't know how to run parent constructors if we don't provide arguments to them

Reply:
When we were using upgradeable, that was bypassed because they don't use constructors